clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2117,14 +2117,23 @@ private ClusterResourceListeners configureClusterResourceListeners(Deserializer<
     private void close(long timeoutMs, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
         AtomicReference<Throwable> firstException = new AtomicReference<>();
+
+        long closeStart = time.milliseconds();
         try {
             if (coordinator != null)
                 coordinator.close(time.timer(Math.min(timeoutMs, requestTimeoutMs)));
         } catch (Throwable t) {
             firstException.compareAndSet(null, t);
             log.error("Failed to close coordinator", t);
         }
-        ClientUtils.closeQuietly(fetcher, "fetcher", firstException);
+        long closeTimeoutLeft = timeoutMs - (time.milliseconds() - closeStart);
+        try {
+            if (fetcher != null)
+                fetcher.close(closeTimeoutLeft);
+        } catch (Throwable t) {
+            firstException.compareAndSet(null, t);
+            log.error("Failed to close Fetcher", t);
```

Comment:
Can we not use closeQuietly?

Reply:
I don't think so since it won't be able to pass the `Timer` class to `fetcher.close`. Maybe we could introduce a new interface called `CloseableWithTimeout`/`TimeoutCloseable` and have `closeQuietly` handle that as well?

Reply:
Or have one that takes a lambda so that the caller can do the `close`. Similar to what we have for Scala.